lib/ui/common/CircularProgress.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:flutter/material.dart';
+import 'dart:math';
+
+class CircularProgress extends StatefulWidget {
+  final Color _lineColor;
+  final double _progress;
+
+  const CircularProgress({Key key, @required double progress, Color lineColor})
+      : this._lineColor = lineColor ?? const Color(0xff24d900),
+        this._progress = progress,
+        assert(progress >= 0 && progress <= 1,
+            'Progress should be between 0 and 1'),
+        super(key: key);
+
+  @override
+  _CircularProgressState createState() => _CircularProgressState();
+}
+
+class _CircularProgressState extends State<CircularProgress> {
+  double progress = 1.0;
+  final double height = 87;
+  final double width = 87;
+  final Color lineColor = Colors.transparent;
+  final double lineWidth = 3;
+  final EdgeInsets innerPadding = const EdgeInsets.all(4.0);
+
+  @override
+  void initState() {
+    super.initState();
+    setState(() {});
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Container(
+        height: height,
+        width: width,
+        child: CustomPaint(
+          foregroundPainter: _CircularPainter(
+              lineColor: lineColor,
+              completeColor: widget._lineColor,
+              completePercent: widget._progress,
+              width: lineWidth),
+          child: Padding(
+            padding: innerPadding,
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _CircularPainter extends CustomPainter {
+  Color lineColor;
+  Color completeColor;
```

Comment:
I think lineColor is better name here, and trackColor above.  This makes me think it changes colour when complete.